src/tests/ui/states/state-needs-payment-info.test.ts

```diff
@@ -0,0 +1,113 @@
+import {} from "svelte";
+import "@testing-library/jest-dom";
+import { render } from "@testing-library/svelte";
+import { describe, test, expect, vi, beforeEach } from "vitest";
+import StateNeedsPaymentInfo from "../../../ui/states/state-needs-payment-info.svelte";
+import {
+  brandingInfo,
+  rcPackage,
+  purchaseResponse,
+} from "../../../stories/fixtures";
+import { SDKEventName } from "../../../behavioural-events/sdk-events";
+import { createEventsTrackerMock } from "../../mocks/events-tracker-mock-provider";
+import { eventsTrackerContextKey } from "../../../ui/constants";
+
+const eventsTrackerMock = createEventsTrackerMock();
+
+const basicProps = {
+  brandingInfo: brandingInfo,
+  purchaseOption: rcPackage.rcBillingProduct.defaultPurchaseOption,
+  productDetails: rcPackage.rcBillingProduct,
+  processing: false,
+  paymentInfoCollectionMetadata: purchaseResponse,
+  onClose: vi.fn(),
+  onContinue: vi.fn(),
+};
+
+const defaultContext = new Map(
+  Object.entries({ [eventsTrackerContextKey]: eventsTrackerMock }),
+);
+
+// vi.mock("@stripe/stripe-js", () => ({
+//   loadStripe: vi.fn().mockResolvedValue({
+//     elements: vi.fn().mockReturnValue({
+//       _elements: [1],
+//       create: vi.fn(),
+//       update: vi.fn(),
+//       mount: vi.fn(),
+//     }),
+//     registerAppInfo: vi.fn(),
+//     confirmPayment: vi.fn().mockResolvedValue({
+//       error: null,
+//     }),
+//     confirmSetup: vi.fn().mockResolvedValue({
+//       error: null,
+//     }),
+//   }),
+//   StripeElements: vi.fn(),
+// }));
```

Comment:
As part of the Stripe Tax implementation, we're splitting the checkout flow in two steps and the clientSecret will no longer needed to render the payment form.

I believe it will be easier to recover these tests when these changes are deployed. 🤞

Reply:
Per our discussion, parking this for the large merge effort. I will add a TODO to the user events branch.